src/main/java/org/openstreetmap/atlas/checks/validation/linear/lines/WaterWayCheck.java

```diff
@@ -0,0 +1,529 @@
+            return true;
+        }
+        if (lineItem instanceof Line)
+        {
+            final Line line = (Line) lineItem;
+            final Location last = line.asPolyLine().last();
+            final Set<LocationItem> one = Iterables.stream(line.getAtlas().nodesAt(last))
+                    .filter(LocationItem.class::isInstance).map(LocationItem.class::cast)
+                    .collectToSet();
+            final Set<LocationItem> two = Iterables.stream(line.getAtlas().pointsAt(last))
+                    .filter(LocationItem.class::isInstance).map(LocationItem.class::cast)
+                    .collectToSet();
+            if (Stream.concat(one.stream(), two.stream())
+                    .anyMatch(this.waterwaySinkTagFilter::test))
+            {
+                return true;
+            }
+        }
+        else if (lineItem instanceof Edge)
+        {
+            return this.waterwaySinkTagFilter.test(((Edge) lineItem).end());
+        }
+        return lineItem.getAtlas()
+                .areasCovering(lineItem.asPolyLine().last(), this.waterwaySinkTagFilter::test)
+                .iterator().hasNext();
+    }
+
+    /**
+     * Check if the last location on a line also happens to be connected to a waterway.
+     *
+     * @param line
+     *            The line to check
+     * @return {@code true} if the line ends on a waterway (the last node is part of a waterway)
+     */
+    public boolean doesLineEndOnWaterway(final LineItem line)
+    {
+        final List<LineItem> waterways = new ArrayList<>();
+        line.getAtlas().lineItemsContaining(line.asPolyLine().last(), this.waterwayTagFilter::test)
+                .forEach(waterways::add);
+
+        waterways.removeIf(line::equals);
+        final Location last = line.asPolyLine().last();
+        return waterways.stream().anyMatch(tLine -> tLine.asPolyLine().contains(last)
+                && !last.equals(tLine.asPolyLine().last()));
+    }
+
+    /**
+     * Check if a location is to the right of a line (e.g., kerbs, cliffs, and oceans)
+     *
+     * @param line
+     *            The line to compare the location to
+     * @param location
+     *            The location
+     * @return {@code true} if the location is to the right of the line. If the location is on the
+     *         line, or to the left of the line, we return {@code false}.
+     */
+    public boolean isRightOf(final PolyLine line, final Location location)
+    {
+        final PolyLine tLine = new PolyLine(location);
+        final Segment closest = line.segments().stream()
+                .min(Comparator.comparingDouble(s -> s.shortestDistanceTo(tLine).asMeters()))
+                .orElse(null);
+        if (closest != null)
+        {
+            final PolyLine testLine = new PolyLine(closest.first(), closest.last(), location);
+            final Angle difference = testLine.headingDifference().orElse(null);
+            if (difference != null && difference.asDegrees() > 0)
```

Comment:
Could be simplified as :  `return difference != null && difference.asDegrees() > 0;`